configuration-resources/messaging/topics.md

```diff
@@ -0,0 +1,224 @@
+
+{% endhint %}
+
+You can also customise all the properties provided by the underlying broker:
+
+```yaml
+    topics:
+        - name: "offset-topic"
+        creation-mode: create-if-not-exists
+        partitions: 1
+        options:
+            consumer.max.poll.records: 10
+            producer.retries: 2
+            replication-factor: 1
+        config:
+            cleanup.policy: compact
+```
+
+You do it by setting the `options` and the `config` configuration parameters.
+
+
+The `config` section is applyed while creating a topic, in the case of Kafka, they become additional configuration options for the topic. Please refer to the Kafka documentation about [topic configurations](https://kafka.apache.org/documentation/#topicconfigs).
+
+The `options` is related to additional properties required to fine tune the client that connects to the topic.
+
+Available options are:
+- `consumer.xxx`: the option xxx is applied to the [Kafka Consumer configuration](https://kafka.apache.org/documentation/#consumerconfigs)
+- `producer.xxx`: the optin xxx is apploied to the [Kafka Producer configuratio](https://kafka.apache.org/documentation/#producerconfigs)
+- `replication-factor`: this is the replication-factor for the topic, when it is created
+
+
+### **Ordering guarantees**
+
+LangStream handles all of the aspects you would have to deal with while building a messaging application, one of the most difficult points is to give strong ordering guarantees.
+For instance if you are implementing a chat bot you really want that the questions and the (streaming, chuncked) answers are processed in the natural order as perceived by the end user.
+This means that in spite of all the processing that is performed thru the pipeline the platform must guarantee that messages are delivered in the expected order.
+
+This is pretty hard, considering that LangStream automatically deals with:
+- temporary failures and retries
+- asyncronous processing
+- (micro) batching
+- scalability
+
+The good thing is that you don't have to care about all of these problems.
+
+But you must be aware that the main way to control the ordering of messages is by means of the **message key**. 
+LangStream guarantees that all the messages with the same key are processed in the same order as they have been written to the topic. If one message enters a retry loop, then all the other messages with the same key are put put on hold until the message is done.
+If the key is null then LangStream is free to process the messages in any order.
+
+
+### **Implicit topics**
+
+The LangStream planner may decide to create additional topics to connect the agents.
+This is because most of the agents may run together in the same kubernetes pod, but under some conditions this is not possible, for instance:
```

Comment:
```suggestion
This is because most of the agents may run together in the same kubernetes pod, but under some conditions this is not possible, for example:
```